### DISCUSSION TITLE
"Subject" in a Use Case Diagram ?

### DISCUSSION THREAD

**Discussion by ebousse** [gaphor/gaphor#1745] [Q&A]
I am willing to re-create in Gaphor this example from the UML specification (page 643, section 18.1.5, UML 2.5.1):

![image](https://user-images.githubusercontent.com/5868014/191471314-5f61bd1a-6b4c-4814-934f-77eb5d7e1570.png)

I don't see how in Gaphor it is possible to draw the _subject_ part of the diagram (here it is the _ATMSystem_ box).

Maybe I am missing something?

thanks in advance!

**Comment by danyeaw** [comment]
Hi @ebousse The subject would be a class in UML. Is it working to add a class to the Use Case diagram?

**Reply by ebousse** [reply]
Yes, if I remember right a "subject" in a UC diagram can be any kind of UML classifier, typically a class or a component (often with the «subsystem» stereotype).

Here is what happens when trying to put a class and a package in a UC diagram in Gaphor:
![image](https://user-images.githubusercontent.com/5868014/191679896-eba88eff-c7b8-4351-a9be-5e79aba3f47c.png)

The class is displayed as if it was inside a class diagram, and the package is displayed as if it was in a package diagram. I could not find a way to make them appear as an open box for the UC diagram.

**Reply by ebousse** [reply]
I just found out that in Gaphor it's possible to drag a UC _inside_ a class or package, to make it "open" itself.

Here is the result:
![Capture d’écran du 2022-09-22 09-13-21](https://user-images.githubusercontent.com/5868014/191682207-e404e22a-391e-41ec-9ae2-6091c95d5e40.png)

Interesting, but still quite different from the expected graphical syntax − especially regarding the component, which keeps its name in the middle!

**Reply by amolenaar** [reply]
> especially regarding the component, which keeps its name in the middle!

This has been fixed in 2.12 :)

We have some [documentation on working with stereotypes](https://gaphor.readthedocs.io/en/latest/stereotypes.html).

**Reply by ebousse** [reply]
> This has been fixed in 2.12 :)

great!

Is there a plan to support the UML official syntax regarding the drawing of a subject within a use case diagram? (ie. as in the example shown in my first message)

> We have some [documentation on working with stereotypes](https://gaphor.readthedocs.io/en/latest/stereotypes.html).

Good to know :) thanks!

And in addition to the possibility to define new stereotypes, are "default" UML stereotypes already defined within Gaphor? (eg. _SubSystem)_

**Comment by danyeaw** [comment]
@ebousse I agree that adding at least some of the stereotypes that are available in the UML Standard Profile makes sense. Would you mind submitting an enhancement request as an issue?

**Reply by amolenaar** [reply]
We can add a few of those to the UML template. It would also be a nice showcase for our stereotype support.
